.github/workflows/run-test-on-pr.yml

```diff
@@ -15,26 +15,72 @@ jobs:
       TOKEN_CONTRACT: ${{ vars.TOKEN_CONTRACT }}
       ORACLE_CONTRACT: ${{ vars.ORACLE_CONTRACT }}
       ENDPOINT: ${{ vars.ENDPOINT }}
-
+    
     steps:
     - name: Checkout repository
       uses: actions/checkout@v4
-    
+ 
+    - name: Debug environment variables
+      run: |
+        echo "TOKEN_CONTRACT: $TOKEN_CONTRACT"
+        echo "ORACLE_CONTRACT: $ORACLE_CONTRACT"
+        echo "ENDPOINT: $ENDPOINT"
+ 
     - name: Use Node.js
       uses: actions/setup-node@v4
       with:
         node-version: '18.x'
         cache: 'npm'
-        
+
     - name: Install dependencies
       run: npm ci
-      
+
+    - name: Run tests againts staging AVS
+      run: |
+        npm test 
+
+  local-test:
```

Comment:
oh, nice, github workflow could already build a docker and test this in action servers?

Then, my suggestion is to separate the tests into two workflows. That way the staging test failure would not affect the local test run.